### ISSUE TITLE
Players get kicked due to the timeout when switching maps

### ISSUE THREAD

**Issue by Dany0** [iamgreaser/iceball#137]
Noticeable with stuff like gen_city.lua which takes a long time to finish. We need to "pause" the time while the switch happens.

**Comment by rakiru** [comment]
Ideally, it would still send pings, but I'm not sure if that would be possible without overly complicating something.

**Comment by BR-** [comment]
Perhaps we should make a pump function that long-running scripts can call periodically to handle the network buffers and whatever else ends up needing to run continuously.
Or, once/if we make an event system, that can just be an event.

**Comment by rakiru** [comment]
Just a call to host.service() or w/e it is would probably be enough, but yeah, if we wrapped that in a generic function that needs called periodically, then anything else can get put in there too, and mods can hook it if they need to add to it.

**Comment by LeComm** [comment]
Something like PySnip's enet reactor.LoopingCall maybe?
Would making it multithreaded cause the typical problems?

**Comment by rakiru** [comment]
No-one sugegsted making it multithreaded, and twisted isn't multithreaded.

**Comment by LeComm** [comment]
Oops I was referring to the ping sending thing.
